### PR TITLE
Added recognition from camera in mobile app

### PR DESCRIPTION
**Pull request by pauliusanuzas**
Also changed the way the camera is opened (now opens in built camera app)

### CODE REVIEW

File: MobileApp/MobileApp.Android/MainActivity.cs [hunk 1]
```diff
@@ -7,7 +7,9 @@
 using Xamarin.Essentials;
 using MobileApp.Views;
 using Android.Graphics;
-
+using Plugin.Permissions;
+using Plugin.Permissions.Abstractions;
+//com.companyname.MobileApp
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp.Android/MainActivity.cs]: redundant comment

File: MobileApp/MobileApp.Android/MainActivity.cs [hunk 2]
```diff
@@ -28,7 +31,7 @@ protected override void OnCreate(Bundle savedInstanceState)
             TabLayoutResource = Resource.Layout.Tabbar;
             ToolbarResource = Resource.Layout.Toolbar;
 
-            CameraPage.cameraButton.Clicked += (sender, args) => { StartActivity(typeof(CameraActivity)); }; //Prisiregistruojama CameraActivity prie mygtuko Clicked evento
+            //CameraPage.cameraButton.Clicked += (sender, args) => { StartActivity(typeof(CameraActivity)); }; //Prisiregistruojama CameraActivity prie mygtuko Clicked evento
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp.Android/MainActivity.cs]: comment

File: MobileApp/MobileApp.Android/MainActivity.cs [hunk 3]
```diff
@@ -48,7 +52,7 @@ public override void OnRequestPermissionsResult(int requestCode, string[] permis
                         else
                         {
                             //Leidimas nesuteiktas
-                            Toast.MakeText(this, "Location permission denied", ToastLength.Short).Show();
+                            Toast.MakeText(this, "Location permission deniededed", ToastLength.Short).Show();
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp.Android/MainActivity.cs]: ? :DD

File: MobileApp/MobileApp/Manager/NavigationManager.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ public void NavigateToMain(IApp app)
         {
             MainPage mainPage = new MainPage();
             app.Main = mainPage;
-            RedirectToLogin(mainPage);
+            //RedirectToLogin(mainPage);
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp/Manager/NavigationManager.cs]: stop it

File: MobileApp/MobileApp/Presenters/CameraPresenter.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    class CameraPresenter : BasePresenter
+    {
+        private CameraPage page;
+        private StoreCameraMediaOptions cameraOptions;
+        private ICarParser parser = new OpenALPRParser();
+        private readonly ITextValidator validator = new LicensePlateValidator();
+        private readonly ICarStatusRequester requester = EPolicijaAPIRequester.GetInstance();
+        private AuthorizationService authorization = AuthorizationService.GetInstance();
+        private LicensePlateService licensePlateService = LicensePlateService.GetInstance();
+        //private Evaluation evaluation = new Evaluation();
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp/Presenters/CameraPresenter.cs]: comment

File: MobileApp/MobileApp/Presenters/CameraPresenter.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+using Models.Users;
+using WebAPIService.EPolicija;
+
+namespace MobileApp.Presenters
+{
+    class CameraPresenter : BasePresenter
+    {
+        private CameraPage page;
+        private StoreCameraMediaOptions cameraOptions;
+        private ICarParser parser = new OpenALPRParser();
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp/Presenters/CameraPresenter.cs]: readonly

File: MobileApp/MobileApp/Presenters/CameraPresenter.cs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        {
+            var photo = await Plugin.Media.CrossMedia.Current.TakePhotoAsync(new StoreCameraMediaOptions());
+            if (photo != null)
+            {
+                CameraPage.cameraImage.Source = ImageSource.FromStream(() => { return photo.GetStream(); });
+                Byte[] imageBytes = ReadFully(photo.GetStream());
+                string plateNumber = await GetPlate(imageBytes);
+                CameraPage.plateLabel.Text = plateNumber;
+
+                System.Console.WriteLine(plateNumber);
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp/Presenters/CameraPresenter.cs]: maybe pop up or something if needed?

File: MobileApp/MobileApp/Presenters/CameraPresenter.cs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+                {
+                    ms.Write(buffer, 0, read);
+                }
+                return ms.ToArray();
+            }
+        }
+
+        private async System.Threading.Tasks.Task<string> GetPlate(Byte[] imageBytes)
+        {
+            string status = "Car not found!";
```
Comment (modestas-jucevicius) [inline MobileApp/MobileApp/Presenters/CameraPresenter.cs:67]: afaik theres an enum?

**Review by modestas-jucevicius — CHANGES_REQUESTED** [review]

File: MobileApp/MobileApp/Presenters/CameraPresenter.cs (hunk 8 above)
Comment (aurisgo) [reply MobileApp/MobileApp/Presenters/CameraPresenter.cs:67]: :100:

Comment (pauliusanuzas) [reply MobileApp/MobileApp/Presenters/CameraPresenter.cs:67]: Pretty sure there is no enum for this. This is when no cars are found in the image, not a car's status.

**Pull request merged**